### PR TITLE
BottomTab dot indicator

### LINKED ISSUE #4910: Possibility to have tab indicator
Is it possible to have a feature like the active indicator on bottom tabs? like a line or something? or maybe it can be achieved through some option (I didn't see any doc about this).

Thank you for your full support

### PR DESCRIPTION
**Pull request by guyca**
This pr adds support for adding a dot indicator to BottomTabs. When both textual badge and dot indicator are defined, the indicator takes precedence, both elements won't be visible at the same time.

image

# API
```js
options: {
  bottomTab: {
    dotIndicator: {
      color: 'green', // default red
      size: 8, // default 8
      visible: true // default false
    }
  }
}
```

# Hiding the indicator
To hide the indicator, call `mergeOptions` with `visible: false`.

```js
Navigation.mergeOptions(this.props.componentId, {
      bottomTab: {
        dotIndicator: { visible: false }
      }
    });
```

Fixes #4910

### CODE REVIEW

File: lib/ios/RNNBasePresenter.m [hunk 1]
```diff
@@ -3,140 +3,159 @@
...
-		readyBlock();
-		readyBlock = nil;
-	}
+    if (readyBlock) {
+        readyBlock();
+        readyBlock = nil;
+    }
 }
 
+- (void)viewDidLayoutSubviews {
```
Comment (yogevbd) [inline lib/ios/RNNBasePresenter.m]: Not needed

**Comment by maleriepace** [comment]
You did it! Thank you! :)

**Comment by emilioheinz** [comment]
This should be available after what version?

**Comment by guyca** [comment]
Yeah, Took a while.. even learned objc for this one 😅

**Pull request merged**